src/modules/dashboard/components/Dialogs/AddExistingSafeDialog/AddExistingSafeDialogForm.tsx

```diff
@@ -179,13 +195,28 @@ const AddExistingSafeDialogForm = ({
   // When selected chain or address changes
   useEffect(() => {
     // If there's a valid address in the address input
-    if (addressTouched && !addressError) {
-      // Get safe data
-      setSafeNotFoundError(undefined);
-      setIsLoadingSafe(true);
-      getSafeData(`${baseURL}api/v1/safes/${address}/`);
+    if (addressTouched && !addressError && isAddress(address)) {
+      /*
+       * Check if safe already exists in colony
+       */
+      const isSafeAlreadyAdded = colonySafes.find(
+        (safe) =>
+          safe?.chainId === selectedChain.value &&
+          safe?.contractAddress === address,
+      );
+
+      if (isSafeAlreadyAdded) {
+        setSafeData('alreadyExists');
+      } else {
+        /*
+         * If not, run api check
+         */
+        setSafeNotFoundError(undefined);
+        setIsLoadingSafe(true);
+        getSafeData(`${baseURL}api/v1/safes/${address}/`);
+      }
     }
-  }, [baseURL, address, addressTouched, addressError]);
+  }, [baseURL, address, addressTouched, addressError, colonySafes]);
```

Comment:
There is a dependancy missing here

<img width="778" alt="Screenshot 2022-07-28 at 21 20 25" src="https://user-images.githubusercontent.com/582700/181609836-c1f80f66-a28f-44a4-87e7-530a40b82c73.png">